### PR TITLE
[Provider] Pinoymovieshub

### PR DESCRIPTION
**Pull request by Jacekun**
ready for merge

### CODE REVIEW

File: PinoyMoviesHub/src/main/kotlin/com/lagradost/PinoyMoviesHub.kt [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+                if (streamLink.isNotBlank()) {
+                    loadExtractor(
+                        url = streamLink,
+                        referer = mainUrl,
+                        callback = callback,
+                        subtitleCallback = subtitleCallback
+                    )
+                }
+            }
+        } catch (e: Exception) {
```
Comment (Blatzar) [inline PinoyMoviesHub/src/main/kotlin/com/lagradost/PinoyMoviesHub.kt:164]: This is already part of the ApiRepository
image

File: PinoyMoviesHub/src/main/kotlin/com/lagradost/PinoyMoviesHub.kt [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+            Pair("Romance", "div#genre_romance"),
+            Pair("Horror", "div#genre_horror"),
+            Pair("Drama", "div#genre_drama"),
+            if (settingsForProvider.enableAdult) Pair("Rated-R 18+", "genre_rated-r") else null
+        )
+        //Log.i(TAG, "Parsing page..")
+        val maindoc = doc.selectFirst("div.module")
+            ?.select("div.content.full_width_layout.full")
+
+        rows.forEach { pair ->
```
Comment (Blatzar) [inline PinoyMoviesHub/src/main/kotlin/com/lagradost/PinoyMoviesHub.kt:43]: mapNotNull would be neat here :pray:

File: PinoyMoviesHub/src/main/kotlin/com/lagradost/PinoyMoviesHub.kt [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+            }
+        }
+        if (day.isBlank()) {
+            day = "01"
+        }
+        if (month.isBlank()) {
+            month = "01"
+        }
+        if (year.isBlank()) {
+            year = if (Build.VERSION.SDK_INT >= 26) {
```
Comment (LagradOst) [inline PinoyMoviesHub/src/main/kotlin/com/lagradost/PinoyMoviesHub.kt:287]: 🤨 providers should not be android version dependant in any way. Make this return null or find another way to return the date

File: PinoyMoviesHub/src/main/kotlin/com/lagradost/PinoyMoviesHub.kt [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+                val ep = Episode(
+                    name = firstA.text(),
+                    data = eplink,
+                    posterUrl = epPoster,
+                    episode = epCount,
+                    season = seasCount,
+                )
+                ep.addDate(parseDateFromString(date))
+                ep
+        } ?: listOf()
```
Comment (LagradOst) [inline PinoyMoviesHub/src/main/kotlin/com/lagradost/PinoyMoviesHub.kt:121]: emptylist exists

File: PinoyMoviesHub/src/main/kotlin/com/lagradost/PinoyMoviesHub.kt [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+
+        val dataUrl = doc.selectFirst("link[rel='shortlink']")
+            ?.attr("href")
+            ?.substringAfter("?p=") ?: ""
```
Comment (LagradOst) [inline PinoyMoviesHub/src/main/kotlin/com/lagradost/PinoyMoviesHub.kt:125]: you should use newMovieResponse and throw an error if no dataUrl is found

Comment (LagradOst) [reply PinoyMoviesHub/src/main/kotlin/com/lagradost/PinoyMoviesHub.kt:125]: it makes it a lot easier to debug "no links found" errors

File: PinoyMoviesHub/src/main/kotlin/com/lagradost/PinoyMoviesHub.kt [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+            ),
+            data = mapOf(
+                Pair("action", action),
+                Pair("post", data),
+                Pair("nume", nume),
+                Pair("type", type)
+            )
+        )
+        //Log.i(TAG, "Response (${doc.code}) => ${doc.text}")
+        AppUtils.tryParseJson<Response?>(doc.text)?.let {
```
Comment (LagradOst) [inline PinoyMoviesHub/src/main/kotlin/com/lagradost/PinoyMoviesHub.kt:181]: ?.embed_url?.let { streamLink-> }

File: PinoyMoviesHub/src/main/kotlin/com/lagradost/PinoyMoviesHub.kt [hunk 7]
```diff
@@ -0,0 +1,299 @@
...
+                    type = tvtype,
+                    posterUrl = image,
+                    year = year,
+                    quality = qual,
+                )
+            }
+        } ?: listOf()
+    }
+
+    private fun parseDateFromString(text: String?): String? {
```
Comment (LagradOst) [inline PinoyMoviesHub/src/main/kotlin/com/lagradost/PinoyMoviesHub.kt:245]: if this format is common for sites then you can implement it into the MainApi helper addDate function 👍

File: PinoyMoviesHub/src/main/kotlin/com/lagradost/PinoyMoviesHub.kt (hunk 3 above)
Comment (Jacekun) [reply PinoyMoviesHub/src/main/kotlin/com/lagradost/PinoyMoviesHub.kt:287]: The site omits the year if its released on current year. That's the only function I find to return the current year. I'll look into finding another way that would support all android versions.

File: PinoyMoviesHub/src/main/kotlin/com/lagradost/PinoyMoviesHub.kt (hunk 7 above)
Comment (Jacekun) [reply PinoyMoviesHub/src/main/kotlin/com/lagradost/PinoyMoviesHub.kt:245]: I'll do it later.

**Comment by Jacekun** [comment]
ready to merge

**Pull request merged**